compiler/rustc_trait_selection/Cargo.toml

```diff
@@ -5,6 +5,7 @@ edition = "2021"
 
 [dependencies]
 # tidy-alphabetical-start
+itertools = "0.12.0"
```

Comment:
the rest of rustc uses `itertools = "0.11"`, can you normalize this? we can upgrade itertools in a separate commit.